AWSAuthSDK/Sources/AWSUserPoolsSignIn/UserPoolsUI/AWSUserPoolNewPasswordRequiredViewController.m

```diff
@@ -93,21 +117,13 @@ -(void) getNewPasswordDetails: (AWSCognitoIdentityNewPasswordRequiredInput *) ne
 }
 /**
  This step completed, usually either display an error to the end user or dismiss ui
+ However [AWSUserPoolsUIOperations startNewPasswordRequired] will be pushing on a new view controller
+ so we will save this error, wait for the next view controller to come on the stack, and then present
+ the error once this this has completed.
  @param error the error if any that occured
  */
 -(void) didCompleteNewPasswordStepWithError:(NSError* _Nullable) error {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        if(error){
-            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:error.userInfo[@"__type"]
-                                                                                     message:error.userInfo[@"message"]
-                                                                              preferredStyle:UIAlertControllerStyleAlert];
-            UIAlertAction *ok = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil];
-            [alertController addAction:ok];
-            [self presentViewController:alertController
-                               animated:YES
-                             completion:nil];
-        }
-    });
+    self.errorOfCompletedTask = error;
```

Comment:
This approach is not ideal, but given that viewControllers are attached to tasks, we are assuming that if this method `didCompleteNewPasswordStepWithError` is called, we are assuming a new view controller will be presented on the screen (via the retry logic in `startNewPasswordRequiredUI`), and therefore, viewDidDisappear is expected to called (and we can present the error at that time).